### ISSUE TITLE
Dilithium speed tests suddenly failing

### ISSUE THREAD

**Issue by randombit** [randombit/botan#3367]
Abruptly, `speed Dilitihum` starts reporting generation of invalid signatures.

First seen in CI for #3366 but now I can replicate it on master on my local machine. And, checking out older revisions, I can replicate it there as well. No idea what is happening.

a) What's wrong with the signatures
b) Why did it not fail in CI before today
c) Why did nobody notice it failing locally before today

Real WTF

**Comment by randombit** [comment]
Verification had the same bug as fixed in #3320 but I do not get why this suddenly manifested, weeks later.

**Comment by randombit** [comment]
OK, a theory on why this wasn't detected earlier.

We only noticed the problem because of the CLI tests. These run `botan speed` but using `--msec=1` to minimize the test runtime.

As a consequence of this, the cli only tended to actually verify a few signatures. If the CI VM was relatively slow, then we might just generate a single signature, verify that signature, then verify an incorrect signature, and then exit. And then perhaps GH upgraded their machines, or just the timing of things changed in some other way, and then the test failed because we started trying to verify more than a single signature during the cli tests.

At least that's the best I can come up with. I did confirm that locally, running the CLI tests failed from a revision last week.

Still a pretty big wtf but I'm not sure what could be checked further wrt post mortem, so closing.
